core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -864,11 +885,21 @@ object ConfigCommand extends Config {
         }
       }
 
+      if (hasEntityName && entityTypeVals.contains(ConfigType.Ip)) {
+        Seq(entityName, ip).filter(options.has(_)).map(options.valueOf(_)).foreach { ipAddress =>
+          if (!Utils.validHostPattern(ipAddress))
```

Comment:
it looks like `max.connections.per.ip` permits hosts in addition to IPs.

```
    val invalidAddresses = maxConnectionsPerIpOverrides.keys.filterNot(address => Utils.validHostPattern(address))
    if (!invalidAddresses.isEmpty)
      throw new IllegalArgumentException(s"${KafkaConfig.MaxConnectionsPerIpOverridesProp} contains invalid addresses : ${invalidAddresses.mkString(",")}")
```
In this case, I'm not sure whether we should go for consistency with the existing IP quota prop or go for more strict enforcement (only IPs permitted).

Reply:
I think that we also permit hosts in our case as `InetAddress.getByName` resolves names that are not IP addresses. The issue is that `Utils.validHostPattern` is just a regex so it does not validate much.

`max.connections.per.ip` is validated in two phases. The code that you outlines is from the `KafkaConfig`. I suppose that we do a lightweight verification there in order to avoid doing any DNS resolution while parsing the config. This is later complemented in the socket server.

```
  private[network] def updateMaxConnectionsPerIpOverride(overrideQuotas: Map[String, Int]): Unit = {
    maxConnectionsPerIpOverrides = overrideQuotas.map { case (host, count) => (InetAddress.getByName(host), count) }
  }
```

In our case, it is important to remain consistent everywhere.